[Qu Tianyue] iP

tianyue58 opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/)) 

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from here.
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉
And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

------------------------------------------------------
If you Java programmer, you can use it to practice Java too. Here's the main method:

     public class Main {
        public static void main(String[] args) {
            Application.launch(MainApp.class, args);
        }
    }

Commit:
Add Gradle support

tianyue58 committed:
add echo feature to Duke.java

tianyue58 committed:
add "Add", "List" features in Duke.java

tianyue58 committed:
add Task.java

tianyue58 committed:
add "mark as done" feature in Duke.java

tianyue58 committed:
add Event.java

tianyue58 committed:
add Deadline.java

tianyue58 committed:
add Todo.java

tianyue58 committed:
handle Todo, Deadline and Event

tianyue58 committed:
create folder Exception and add two exception classes

tianyue58 committed:
handle exceptions in Duke.java

tianyue58 committed:
add delete function

tianyue58 committed:
move two exceptions class outside Exception folder

tianyue58 committed:
add test cases in input.txt and Expected.TXT

tianyue58 committed:
update EXPECTED.TXT by directly copying from ACTUAL.TXT

tianyue58 committed:
delete trailing white space on L20

tianyue58 committed:
update with original expected input

tianyue58 committed:
update with original expected.txt

tianyue58 committed:
remove trailing whitespace on L20

tianyue58 committed:
delete trailing white space

tianyue58 committed:
add new line using "\n"

tianyue58 committed:
add new line at the end by pressing ENTER

tianyue58 committed:
delete "\n" in print statement

tianyue58 committed:
Implement Save

tianyue58 committed:
Change date format for deadline and event

tianyue58 committed:
Merge branch 'branch-Level-8'

tianyue58 committed:
Make previous code more OOP

tianyue58 committed:
Divide classes into packages

tianyue58 committed:
Add Duke.java to duke package

tianyue58 committed:
Add Storage to util package

tianyue58 committed:
Add JUnit tests

tianyue58 committed:
Add manifest file for building jar

tianyue58 committed:
Add JavaDoc comments

tianyue58 committed:
Tweak the code to comply with a coding standard

tianyue58 committed:
Add JavaDoc comments

tianyue58 committed:
Tweak code to comply with coding standards

tianyue58 committed:
Implement Level 9

tianyue58 committed:
Merge branch 'branch-A-JavaDoc'

tianyue58 committed:
Merge branch 'branch-Level-9'

tianyue58 committed:
Add DukeIndexException and FindCommand

tianyue58 committed:
Fix bug in TaskList

tianyue58 committed:
Fix bug in event

tianyue58 committed:
Fix bugs in reading from file

tianyue58 committed:
Merge branch 'add-gradle-support' of https://github.com/tianyue58/ip into add-gradle-support

tianyue58 committed:
Add a GUI to Duke

tianyue58 committed:
Add comments according to Java coding standard

NayLin-H99 commented on src/main/java/duke/BotOutput.java:
```diff
@@ -0,0 +1,22 @@
+    public boolean isExit() {
+        return isExit;
+    }
+}
```
Do leave a line at the end your file! 😄

tianyue58 replied:
Okay, thanks for the reminder!

NayLin-H99 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,15 @@
+/**
+ * Command class is an abstract class that different command classes inherit from.
+ */
+public abstract class Command {
```
If you don't have any constructors or any concrete methods under `Command`, perhaps consider changing it to an interface?

tianyue58 replied:
Okay, thanks for the suggestion!

NayLin-H99 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,42 @@
+ * DeleteCommand class which handles the logic of deleting tasks.
+ */
+public class DeleteCommand extends Command {
+    private final String fullCommand;
```
You can consider adding Javadocs for class members as stated under Java coding standards. 😄 
![image](https://user-images.githubusercontent.com/77244987/131902924-e9151807-78c5-4c9d-a77d-e7f8036faf37.png)

tianyue58 replied:
Okay! 😊

NayLin-H99 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,57 @@
+import duke.task.Todo;
+import duke.util.Ui;
+import duke.util.Storage;
+import java.util.ArrayList;
```
Do you want to include a linebreak before this for consistency with your other classes?

tianyue58 replied:
Yup, you're right. Thanks for pointing this out!

NayLin-H99 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,101 @@
+     * @param description Description of a deadline.
+     * @param by Due time of a deadline.                   
+     */
+    public Deadline(String isDone, String description, String by) {
```
You are using `isDone` string as if it is a boolean type. Perhaps consider using a boolean instead?

choonyongchan replied:
Agreed.

NayLin-H99 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,103 @@
+     * @return Task description.
+     * @throws UnclearInstructionException  If the given command cannot be properly extracted or has empty description.
+     */
+    public static String extractTaskDescription(String text) throws UnclearInstructionException {
```
I think you can perhaps consider doing String parsing under Parser? 😃

tianyue58 replied:
Yup, you're right. Thanks for pointing this out!

NayLin-H99 commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,38 @@
+package duke.util;
+
+import duke.command.*;
```
Try to configure your IDE to not optimise imports. Imports (at least in this module) should always be explicitly stated. 
![image](https://user-images.githubusercontent.com/77244987/131905115-1859339f-a29d-4bde-8f73-863a6f7c4f80.png)

tianyue58 replied:
Ohh yeah, I didn't notice IntelliJ automatically changes my imports to *. Thanks for pointing it out!

NayLin-H99 commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,130 @@
+package duke.util;
+
+import duke.exceptions.UnclearInstructionException;
+import duke.exceptions.UnknownException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
```
Very neat imports! 😃

NayLin-H99 commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,138 @@
+        outputs.add("Got it. I've added this task:");
+        outputs.add(task.toString());
+        list.add(task);
+        if (list.size() == 1) {
+            outputs.add("Now you have 1 task in the list.");
+        } else {
+            outputs.add(String.format("Now you have %d tasks in the list.", 
+                    list.size()));
+        }
```
It is really meticulous of you to take note of the plural form! Good job! 👍

tianyue58 replied:
Thank you!

NayLin-H99 commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,16 @@
+
+public class DeadlineTest {
+    @Test
+    public void deadline_toString_success() {
```
Perhaps you can consider adding more tests to check for the exceptions thrown on invalid output?

tianyue58 replied:
Okay, thanks for the suggestion!

NayLin-H99 reviewed (commented):
I think it is a really good job done overall! Really good on you for being meticulous and adding a personal touch to Duke! 😃

choonyongchan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,101 @@
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import javafx.scene.image.Image;
```
I like the consistency of the import statements thus far. Perhaps you could remove the new lines between 'import javafx' statements to make the import statements perfect?

tianyue58 replied:
Yup, you're right. Thanks for pointing this out!

choonyongchan commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Override
+    public ArrayList<String> execute(TaskList taskList, Ui ui, Storage storage) {
+        
```
Perhaps this new line can be removed to maintain consistency?

tianyue58 replied:
Yup, thanks for pointing this out! 😃

choonyongchan commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,130 @@
+                if (task_type.equals("T")) {
+                    task = new Todo(itemDetails[2], itemDetails[1]);
+
+                } else if (task_type.equals("D")) {
```
Perhaps this new line can also be removed to maintain consistency? :)

tianyue58 replied:
Yup, you're right. Thanks for pointing this out!😊

choonyongchan reviewed (commented):
Overall, nicely organised and done. Only some nits to iron out. Great job.

tianyue58 committed:
Fix java coding standard and improve code quality based on peer review

tianyue58 committed:
Improve bot UI

YeluriKetan commented on src/main/java/duke/BotOutput.java:
```diff
@@ -0,0 +1,22 @@
+ * BotOutput class encapsulates responses from duke.
+ */
+public class BotOutput {
+    private final String botOutput;
```
You could perhaps name this constant as BOT_OUTPUT to better follow coding standards?

tianyue58 replied:
Yes, I agree 👍

YeluriKetan commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,138 @@
+ * operations relating to the tasks.
+ */
+public class TaskList {
+    private List<Task> list;
```
A preferred naming convention for collection of objects according to the coding standard is to use Plural form. Could maybe name the `list` as `tasks` or an alternative of your choice?

tianyue58 replied:
Yea you're right. Thanks for pointing it out! 👍

YeluriKetan reviewed (commented):
LGTM. Coding standard followed with very minor violations, JavaDocs are detailed and well done. Most other feedback points have already been mentioned by fellow students and reviewers. 
Great job and good luck!!

tianyue58 committed:
Update build.gradle

Ensure build.gradle file contains the correct values w.r.t. the Shadow plugin e.g., Launcher class

tianyue58 committed:
Add assertions

Use assert feature to document important assumptions that should hold at various points in the code.

For example, the input for some functions should not be null. With assertion, it helps to avoid causing
Null Pointer Exceptions in some cases.

tianyue58 committed:
Remove redundant code

tianyue58 committed:
Merge pull request #2 from tianyue58/branch-A-CodeQuality

Remove redundant code

tianyue58 committed:
Merge branch 'master' into branch-A-Assertions

* master:
  Remove redundant code

tianyue58 committed:
Merge pull request #1 from tianyue58/branch-A-Assertions

Add assertions

tianyue58 committed:
Add extension for friendlier syntax

tianyue58 committed:
Fix coding style violation

tianyue58 committed:
Fix error in logic

Allow the bot to do exception handling correctly and clean up redundant code.

tianyue58 committed:
Improve the GUI

tianyue58 committed:
Add Ui.png

tianyue58 committed:
Add product name in the title bar of the window and disable resizing window option

tianyue58 committed:
Add welcome screen image

tianyue58 committed:
Add FIND screenshot

tianyue58 committed:
Add a User Guide

tianyue58 committed:
Set theme jekyll-theme-cayman

tianyue58 committed:
Update code according to java coding standard

tianyue58 committed:
Merge branch 'master' of https://github.com/tianyue58/ip

* 'master' of https://github.com/tianyue58/ip:
  Set theme jekyll-theme-cayman

tianyue58 committed:
Update UG

tianyue58 committed:
Change jar file link in README.md

tianyue58 committed:
Edit README.md for minor error in quick start instruction

tianyue58 committed:
Improve error handling

tianyue58 committed:
Update README.md

tianyue58 committed:
Update README.md

tianyue58 committed:
Update README.md

tianyue58 committed:
Add JUnit Test
